private/protocol/timestamp_test.go

```diff
@@ -0,0 +1,54 @@
+// +build go1.7
+
+package protocol_test
+
+import (
+	"github.com/aws/aws-sdk-go/private/protocol"
+	"testing"
+	"time"
+)
+
+func TestFormatTime(t *testing.T) {
+	cases := map[string]struct {
+		formatName, expectedOutput string
+		input time.Time
+	}{
+		"UnixTest1": {formatName: protocol.UnixTimeFormatName,expectedOutput: "946845296" , input: time.Date(2000, time.January, 2, 20, 34, 56, .123e9, time.UTC)},
+		"ISO8601Test1": {formatName: protocol.ISO8601TimeFormatName,expectedOutput: "2000-01-02T20:34:56Z" , input: time.Date(2000, time.January, 2, 20, 34, 56, .123e9, time.UTC)},
+		"RFC822Test1": {formatName: protocol.RFC822TimeFormatName,expectedOutput: "Sun, 02 Jan 2000 20:34:56 GMT" , input: time.Date(2000, time.January, 2, 20, 34, 56, 0, time.UTC)},
```

Comment:
It would be helpful to format these cases onto multiple lines for easier readability. Also need to make sure `gofmt` is being run for the code.